website/client/components/groups/startQuestModal.vue

```diff
@@ -177,13 +182,14 @@ export default {
       let groupId = this.group._id || this.user.party._id;
 
       const key = this.selectedQuest;
-      const response = await this.$store.dispatch('guilds:inviteToQuest', {groupId, key});
-      const quest = response.data.data;
-
-      if (this.$store.state.party.data) this.$store.state.party.data.quest = quest;
-
-      this.loading = false;
-
+      try {
+        const response = await this.$store.dispatch('guilds:inviteToQuest', {groupId, key});
+        const quest = response.data.data;
+
+        if (this.$store.state.party.data) this.$store.state.party.data.quest = quest;
+      } finally {
```

Comment:
The problem was that `guilds:inviteToQuest` threw an error when an invalid quest key was passed, which then logically aborts the rest of the method call. As a result, `this.loading` was never reset, so the quest button remained disabled. 

The `finally` ensures that the button will always re-enable after the call returns, even if it returns an error. I specifically left the `emit` because I figured that you probably want the dialog to stay open if the invite fails so that you can try another quest.

Reply:
Ok, but if you remove the code from the final block and move it outside, if an error happens in the try block it won't throw so it'll keep going, right?

```
try {
...
} catch (e) {} // empty catch block

... other code ...

```

anyway, not a big problem, I was worried about finally not being supported but looks like it's not a problem

Reply:
Aaaah, that makes more sense. I thought you meant removing it altogether, but you're talking about replacing `finally` with `catch`. You're right that either will work fine. I suppose I picked this route because the split between `this.loading` (which you always want to trigger) and `emit` (which you only want to trigger on a successful call) was more obvious with this implementation.